paasta_tools/kubernetes/bin/paasta_secrets_sync.py

```diff
@@ -245,63 +283,95 @@ def sync_secrets(
-                secret_signature = secret_provider.get_secret_signature_from_data(
-                    secret_data
-                )
-                if secret_signature:
-                    kubernetes_secret_signature = get_kubernetes_secret_signature(
-                        kube_client=kube_client,
-                        secret=secret,
-                        service=service,
-                        namespace=namespace,
+                    secret_signature = secret_provider.get_secret_signature_from_data(
+                        json.load(secret_file)
                     )
-                    if not kubernetes_secret_signature:
-                        log.info(
-                            f"{secret} for {service} not found in {namespace}, creating"
-                        )
-                        try:
-                            create_secret(
-                                kube_client=kube_client,
-                                secret=secret,
-                                service=service,
-                                secret_provider=secret_provider,
-                                namespace=namespace,
-                            )
-                        except ApiException as e:
-                            if e.status == 409:
-                                log.warning(
-                                    f"Secret {secret} for {service} already exists in {namespace}"
-                                )
-                            else:
-                                raise
-                        create_kubernetes_secret_signature(
-                            kube_client=kube_client,
-                            secret=secret,
-                            service=service,
-                            secret_signature=secret_signature,
-                            namespace=namespace,
-                        )
-                    elif secret_signature != kubernetes_secret_signature:
-                        log.info(
-                            f"{secret} for {service} in {namespace} needs updating as signature changed"
-                        )
-                        update_secret(
-                            kube_client=kube_client,
-                            secret=secret,
-                            service=service,
-                            secret_provider=secret_provider,
-                            namespace=namespace,
-                        )
-                        update_kubernetes_secret_signature(
-                            kube_client=kube_client,
-                            secret=secret,
-                            service=service,
-                            secret_signature=secret_signature,
-                            namespace=namespace,
-                        )
-                    else:
-                        log.info(f"{secret} for {service} in {namespace} up to date")
+                update_k8s_secret(
```

Comment:
should this be renamed so its a little clearer without reading this function's body that it will create the secret if it doesn't exist?
```suggestion
                create_or_update_k8s_secret(
```